cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsMetadataCollector.java

```diff
@@ -215,22 +221,14 @@ private void addKnownInstance(CloudInstance cloudInstance, Multimap<String, Inst
 
     @Override
     public InstanceStoreMetadata collectInstanceStorageCount(AuthenticatedContext ac, List<String> instanceTypes) {
-        AwsCredentialView credentialView = new AwsCredentialView(ac.getCloudCredential());
-        AmazonEc2Client amazonEC2Client = awsClient.createEc2Client(credentialView);
-        DescribeInstanceTypesRequest request = new DescribeInstanceTypesRequest().withInstanceTypes(new HashSet<>(instanceTypes));
-        try {
-            DescribeInstanceTypesResult result = amazonEC2Client.describeInstanceTypes(request);
-            InstanceStoreMetadata instanceStoreMetadata = new InstanceStoreMetadata();
-            for (InstanceTypeInfo instanceTypeInfo: result.getInstanceTypes()) {
-                instanceStoreMetadata.addInstanceStoreCountToInstanceType(instanceTypeInfo.getInstanceType(),
-                        instanceTypeInfo.getInstanceStorageSupported() ?
-                        instanceTypeInfo.getInstanceStorageInfo().getDisks() != null ?
-                        instanceTypeInfo.getInstanceStorageInfo().getDisks().stream().map(DiskInfo::getCount).mapToInt(Integer::intValue).sum() : 0 : 0);
-            }
-            return instanceStoreMetadata;
-        } catch (Exception e) {
-            LOGGER.warn("Failed to describe instance types: {}", instanceTypes, e);
-            throw new CloudConnectorException(e.getMessage(), e);
-        }
+        Location location = ac.getCloudContext().getLocation();
+        CloudVmTypes cloudVmTypes = awsPlatformResources.virtualMachines(ac.getCloudCredential(), location.getRegion(), Map.of());
```

Comment:
It's a little bit defensive programming, but as I checked that AwsPlatformResources service doesn't really care about exceptions, so I would not remove the exception handling from the original code 😸